torch/autograd/_functions/tensor.py

```diff
@@ -591,6 +591,16 @@ def backward(self, grad_output):
         return grad_input
 
 
+def reverse_cumsum(x, dim):
+    ret = torch.cumsum(-x, dim=dim)
+
+    end_idx = ret.size(dim) - 1
+    ret_sum = ret.narrow(dim, end_idx, 1)
```

Comment:
As a small comment, this will also affect the current behaviour of Cumsum when merged (the clone is missing in master for Cumsum as well).